### DISCUSSION TITLE
CMAKE errors while installing Openroad

### DISCUSSION THREAD

**Discussion by mdzaki-git** [The-OpenROAD-Project/OpenROAD#2972] [Q&A] [answered]
HI All,

After installing the dependencies we re stuck with cmake errors and would appreciate your help in fixing them. Attaching the error snippet for reference.
<img width="1388" alt="Cmake Error" src="https://user-images.githubusercontent.com/68725332/223042193-39b47832-730e-4716-abce-421e7a44dfc8.png">

**Comment by vijayank88** [comment]
attach log `build_openroad.log`

**Reply by vijayank88** [reply]
Also share instruction you followed for the installation

**Comment by mdzaki-git** [comment]
[openroad_build.log](https://github.com/The-OpenROAD-Project/OpenROAD/files/10897535/openroad_build.log)

**Reply by vijayank88** [reply]
@vvbandeira Can you please check this issue?

**Reply by vijayank88** [reply]
Problem in getting CMake:
```
CMake Warning at CMakeLists.txt:105 (message):
  OpenROAD git describe failed, using sha1 instead
 ```

**Reply by vvbandeira** [reply]
@vijayank88, this warning is not a problem with CMake; this indicates that the folder is not a git repo. @mdzaki-git, can you make sure that the clone ended fine? Maybe check the output of `git status --long`, the output of this command should look something like this:

```
> git status --long
On branch master
Your branch is up to date with 'origin/master'.
nothing to commit, working tree clean
```

Also, try to do a clean build and in case of any errors upload the new log file.
```
./etc/Build.sh -clean
```

**Comment by mdzaki-git** [comment]
Following the install instruction mentioned on git hub.
https://github.com/The-OpenROAD-Project/OpenROAD

**Comment by mdzaki-git** [comment]
Hi Vitor,

Recloned the repository and tried again, getting below errors now.

**Comment by mdzaki-git** [comment]
[openroad_build.log](https://github.com/The-OpenROAD-Project/OpenROAD/files/10899446/openroad_build.log)

**Comment by vvbandeira** [comment, marked as answer]
Looks like you missed the `--recursive` when cloning. Please run

```
git submodule update --init
```

In the future you should use `git clone --recursive` to also clone the submodules `abc` and `sta`.

**Comment by mdzaki-git** [comment]
Hi All,

Thanks it works now.
